test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs

```diff
@@ -0,0 +1,488 @@
+                {
+                    var odataWriter = await messageWriter.CreateODataResourceWriterAsync(peopleSet, personType);
+                    var entry = this.CreatePersonEntryWithoutSerializationInfo();
+
+                    entry.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "Parsen", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+                    await odataWriter.WriteStartAsync(entry);
+                    await odataWriter.WriteEndAsync();
+                    var result = await TestsHelper.ReadStreamContentAsync(responseMessageWithoutModel.GetStream());
+                    Assert.Contains("\"PersonId\":-5", result);
+                    if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                    {
+                        // no metadata does not write odata.metadata
+                        Assert.Contains("$metadata#Parsen/$entity", result);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Change serialization info value after WriteStart
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task ChangeSerializationInfoAfterWriteStart(string mimeType)
+        {
+            var settings = new ODataMessageWriterSettings();
+            settings.ODataUri = new ODataUri() { ServiceRoot = _baseUri };
+
+            var responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceWriterAsync();
+                var entry = this.CreatePersonEntryWithoutSerializationInfo();
+
+                entry.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "People", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+                await odataWriter.WriteStartAsync(entry);
+                entry.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "Parsen", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+                await odataWriter.WriteEndAsync();
+
+                Stream stream = await responseMessageWithoutModel.GetStreamAsync();
+
+                // nometadata option does not write odata.metadata in the payload
+                if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                {
+                    Assert.Contains("People/$entity", (await TestsHelper.ReadStreamContentAsync(stream)));
+                }

```

Comment:
Consider making this a parameter. Note that there's a different between:
- not asserting that some condition is true
- asserting that some condition is not true

In most of your tests you're doing the former, which does not give us any insight about the latter. Because we have behaviour that's not defined or acknowledged by this test, and we don't know if that's intentional or not. If nometadata leads to odata.metadata not being written and that's expected, then why not assert that?

Adding a boolean parameter to this test would allow having a single assert statement for all scenarios and we could look at the parameters to tell when it's expected to be true or false. Or we could have separate tests if those scenarios are different enough.